frontend/javascripts/dashboard/dataset/sample_datasets_modal.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+import { Spin, Button, Modal, List } from "antd";
+import React, { useState, useEffect } from "react";
+import _ from "lodash";
+
+import {
+  getSampleDatasets,
+  triggerSampleDatasetDownload,
+  getDatastores,
+} from "admin/admin_rest_api";
+import { useInterval } from "libs/react_helpers";
+import { handleGenericError } from "libs/error_handling";
+import type { APIDataStore, APISampleDataset } from "admin/api_flow_types";
+
+type Props = {
+  destroy: () => void,
+  onOk?: () => any,
+  organizationName: string,
+};
+
+function useDatastores(): [Array<APIDataStore>] {
```

Comment:
I imagine that these ten lines of code will be useful quite a lot in the future (independent of entity type, of course). We don't have to do it now, but I'd think that something like `const datastores = useAsyncGet(getDataStores, [])` would be a good abstraction (naming is debatable).

Reply:
I created a generic `useFetch` method and included a third parameter for the dependencies (for useEffect) :)